Lombiq.Hosting.Tenants.Management/Service/JsonConfigurationParser.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text.Json;
+
+namespace Lombiq.Hosting.Tenants.Management.Service;
+
+public class JsonConfigurationParser
+{
+    private readonly Dictionary<string, string> Data = new(StringComparer.OrdinalIgnoreCase);
+    private readonly Stack<string> Paths = new();
+
+    public IDictionary<string, string> ParseConfiguration(string inputJson)
+    {
+        var jsonDocumentOptions = new JsonDocumentOptions
+        {
+            CommentHandling = JsonCommentHandling.Skip,
+            AllowTrailingCommas = true,
+        };
+
+        using var doc = JsonDocument.Parse(inputJson, jsonDocumentOptions);
+        if (doc.RootElement.ValueKind != JsonValueKind.Object)
+        {
+            throw new FormatException("Format Exception:" + doc.RootElement.ValueKind);
```

Comment:
Use exceptions that explain the issue in sufficient detail, and use the spelling of English sentences, in all three cases.